daemon/daemon_unix.go

```diff
@@ -584,11 +583,10 @@ func cgroupDriver(cfg *config.Config) string {
 // getCD gets the raw value of the native.cgroupdriver option, if set.
 func getCD(config *config.Config) string {
 	for _, option := range config.ExecOptions {
-		key, val, err := parsers.ParseKeyValueOpt(option)
-		if err != nil || !strings.EqualFold(key, "native.cgroupdriver") {
-			continue
+		key, val, ok := strings.Cut(option, "=")
+		if ok && strings.EqualFold(strings.TrimSpace(key), "native.cgroupdriver") {
```

Comment:
fwiw; I really want to look at some of these to be more strict altogether (not case-insensitive, and less manipulation of values like trimming spaces etc); I kept the current behavior for now though.